### PR TITLE
Fix downgrade network error causing UI crash

### PR DESCRIPTION
**Pull request by joshenlim**

### CODE REVIEW

**Review by alaister — APPROVED** [review]
LGTM!

File: apps/studio/components/interfaces/Organization/BillingSettings/Subscription/PlanUpdateSidePanel.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ const PlanUpdateSidePanel = () => {
   const { data: membersExceededLimit } = useFreeProjectLimitCheckQuery({ slug })
   const { mutate: updateOrgSubscription, isLoading: isUpdating } = useOrgSubscriptionUpdateMutation(
     {
-      onSuccess: () => {
+      onSuccess: (data) => {
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/interfaces/Organization/BillingSettings/Subscription/PlanUpdateSidePanel.tsx:84]: oh nope! was debugging something haha

File: apps/studio/components/interfaces/Integrations/IntegrationConnection.tsx [hunk 2]
```diff
@@ -49,6 +44,8 @@ const IntegrationConnectionItem = forwardRef<HTMLLIElement, IntegrationConnectio
     const onConfirm = useCallback(async () => {
       try {
         await onDeleteConnection(connection)
+      } catch (error) {
+        // [Joshen] No need for error handler
```
Comment (ivasilov) [inline apps/studio/components/interfaces/Integrations/IntegrationConnection.tsx:48]: In 2 of the 3 uses of the this component, there's a throw in the promise.
https://github.com/supabase/supabase/blob/master/apps/studio/components/interfaces/Organization/IntegrationSettings/IntegrationSettings.tsx#L61
https://github.com/supabase/supabase/blob/master/apps/studio/components/interfaces/Settings/Integrations/GithubIntegration/GithubSection.tsx#L79

File: apps/studio/components/interfaces/Settings/Vault/Secrets/AddNewSecretModal.tsx [hunk 3]
```diff
@@ -57,28 +57,21 @@ const AddNewSecretModal = ({ visible, onClose }: AddNewSecretModalProps) => {
...
-      if (!res.error) {
-        toast.success(`Successfully added new secret ${values.name}`)
-        onClose()
-      } else {
-        toast.error(`Failed to add secret ${values.name}: ${res.error.message}`)
-      }
+      toast.success(`Successfully added new secret ${values.name}`)
+      onClose()
+    } catch (error: any) {
+      // [Joshen] No error handler required as they are all handled within the mutations already
```
Comment (ivasilov) [inline apps/studio/components/interfaces/Settings/Vault/Secrets/AddNewSecretModal.tsx:74]: I'm not 100% sure, but I think when using `mutateAsync`, `onError` are skipped.

Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/Vault/Secrets/AddNewSecretModal.tsx:74]: tested this one locally - can verify that onError from the mutateAsyncs will be run

File: apps/studio/components/interfaces/Integrations/IntegrationConnection.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/IntegrationConnection.tsx:48]: ahhhh good catch - they shouldn't need to throw, i'll just make them return (like how we handle projectref checks before calling a mutation)

Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/IntegrationConnection.tsx:48]: okay pushed a change! 🙏

**Review by ivasilov — APPROVED** [review]

**Pull request merged**